account_update.php

```diff
@@ -100,8 +99,8 @@
 # Do not update email for a user verification
 if( !( $t_ldap && config_get( 'use_ldap_email' ) )
 	&& !$t_account_verification ) {
-	if( $f_email != user_get_email( $t_user_id ) ) {
-		$t_update_email = $f_email;
+	if( !is_blank( $f_email ) && $f_email != user_get_email( $t_user_id ) ) {
```

Comment:
Have a look at the discussion at https://www.mantisbt.org/bugs/view.php?id=21737
This is to prevent users from removing their email address.